src/middleware/socket.js

```diff
@@ -232,81 +180,86 @@ const chatMiddleware = store => next => action => {
 
     if (history.length) {
-      const lastMessage = history[history.length - 1];
-      updateForAnySpecialMessage({ lastMessage, store });
-      store.dispatch(setHistory(history));
-    } else {
-      store.dispatch(setHistory([{ role: roles.assistant, content: config.aiProfile.initialMessage, time: new Date() }]));
-      handleMessageSending({
-        role: roles.assistant,
-        term: getQueryParam(window.location.search, 'utm_chat'),
-        user_id: meta.cid,
-        message: config.aiProfile.initialMessage,
-        region: meta.region,
-      });
+      store.dispatch(setExistingHistory(history));
+      return;
     }
+
+    const id = uid();
+    config.aiProfile.initialMessage.map(content => store.dispatch(fillAssistantHistoryData({ id, content })));
+    store.dispatch(setResponseFormVisibility(config.aiProfile.initialMessage));
+    handleMessageSending({
+      role: roles.assistant,
+      term: getQueryParam(window.location.search, 'utm_chat'),
+      user_id: meta.cid,
+      message: 'initial message',
+      initialMessageContent: config.aiProfile.initialMessage,
+      initialMessageId: id,
+      region: meta.region,
+    });
   });
 
-  socket.on(events.streamStart, () => {
+  // remove => meant to be for socket mock
+  let idx = 0;
+  let messageUuid = faker.string.uuid();
+
+  let mockedDataForEachStream = streamMock;
+
+  const resetUUID = () => {
+    messageUuid = faker.string.uuid();
+    mockedDataForEachStream = mockedDataForEachStream.map(i => ({ ...i, id: messageUuid }));
+  };
+
+  socket.on(events.streamStart, (data) => {
+    // remove => meant to be for socket mock
+    resetUUID();
+
+    store.dispatch(setIsStreaming(true));
     store.dispatch(resetIsLoading());
     store.dispatch(resetOutgoing());
-    store.dispatch(setIncoming());
     store.dispatch(resetError());
-  });
 
-  socket.on(events.streamData, ({ chunk, errors, question_id, answer_id }) => {
-    const { chat, config } = store.getState();
-    const { textToParse, incoming } = chat;
-    const link = constructLink(textToParse) || constructLink(incoming?.content) || constructLink(chunk);
+    store.dispatch(fillAssistantHistoryData({ id: mockedDataForEachStream[idx].id, content: mockedDataForEachStream[idx] }));
 
-    if (errors.length && !chat.error) {
-      store.dispatch(setError(errors[0]));
-    }
+    // remove => meant to be for socket mock
+    idx += 1;
+  });
 
-    if (link) {
-      store.dispatch(setLink({
-        isVisible: true,
-        href: link,
-        text: config.translations.mealButton
-      }));
-      store.dispatch(setResponseFormVisibility({ isFormVisible: false }));
-    }
+  socket.on(events.streamData, (data) => {
+    const { chat } = store.getState();
 
-    if (chunk.includes('[')) {
-      store.dispatch(setTextToParse(chunk));
-      return;
+    const current = mockedDataForEachStream[idx];
+    if (current) {
+      if (current.type === 'email') store.dispatch(setIsEmailFormVisible(true));
```

Comment:
@IlievaDayana It is better to be consistent and always use curly braces for if statement body like the format used on line 237 below.

`if (current.type === 'email') {
   store.dispatch(setIsEmailFormVisible(true));
}`